packages/dart/noports_core/lib/src/srv/srv_impl.dart

```diff
@@ -630,69 +637,111 @@ class SrvImplDart implements Srv<SocketConnector> {
     return socketConnector;
   }
 
-  Future _handleMultiConnectRequest(
+  Future<void> _handleMultiConnectRequest(
+    SocketConnector sc,
+    InternetAddress relayAddress,
+    DataTransformer? encrypter,
+    DataTransformer? decrypter,
+  ) async {
+    logger.info('_runDaemonSideMulti'
+        ' Control socket received connect request - '
+        ' creating new socketToSocket connection');
+
+    InternetAddress localAddress = await resolveRequestedLocalHost();
+
+    // First, connect to the relay.
+    // If it fails, return, as we can't do anything more
+    late Socket sideBSocket;
+    late Side sideB;
+    logger.info(
+        'socket_connector: Connecting side B (relay - $relayAddress:$streamingPort)');
+    try {
+      sideBSocket = await Socket.connect(relayAddress, streamingPort);
+      sideB = Side(sideBSocket, false, transformer: decrypter);
+      unawaited(sideBSocket.done
+          .then((v) => logger.info('relay socket done'))
+          .catchError(
+              (err) => logger.warning('relay socket done with error $err')));
+      if (rvdAuthString != null) {
+        logger.info('_runDaemonSideMulti authenticating'
+            ' new socket connection to relay');
+        sideBSocket.writeln(rvdAuthString);
+        await sideBSocket.flush();
+      }
+    } catch (e) {
+      logger.shout(
+          'Failed to connect to relay ($relayAddress:$streamingPort) with error : $e');
+      return;
+    }
+
+    // Now, connect to the local host:port
+    // If it fails, we need to close the socket connection to the relay
```

Comment:
When we close the socket connection to the relay, the relay will close its connection to the client.